### PR TITLE
added a test block to series.py

### PR DESCRIPTION
**Pull request by cthaehtruthteller**

### CODE REVIEW

**Comment by cthaehtruthteller** [comment]
This request is for series.py and slicing.py

File: students/thecthaeh/Lesson02/series.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    if n == 0:
+        return 2
+        
+    elif n == 1:
+        return 1
+        
+    return lucas(n - 2) + lucas(n - 1)
+
+
+def sum_series(n, index0 = 0, index1 = 1):
```
Comment (charlesrobison) [inline students/thecthaeh/Lesson02/series.py]: You'll want to remove the white space around the assignment operators here - PEP8 formatting

File: students/thecthaeh/Lesson02/series.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    
+    # test that sum_series works for arbitrary initial values
+    assert sum_series(0, 3, 2) == 3
+    assert sum_series(1, 3, 2) == 2
+    assert sum_series(2, 3, 2) == 5
+    assert sum_series(3, 3, 2) == 7
+    assert sum_series(4, 3, 2) == 12
+    assert sum_series(5, 3, 2) == 19
+    
+    print("tests passed")
```
Comment (charlesrobison) [inline students/thecthaeh/Lesson02/series.py]: Works well!

File: students/thecthaeh/Lesson03/list_lab.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+print(fruit_list)
+
+add_fruit = input("Please enter a fruit of your choosing. ")
+
+fruit_list.append(add_fruit)
+
+print(fruit_list)
+
+num_response = input("Enter a number. ")
```
Comment (charlesrobison) [inline students/thecthaeh/Lesson03/list_lab.py]: len() is a good option here to let the user know how long the list is.

File: students/thecthaeh/Lesson03/list_lab.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+f = fruit_list[:]
+
+for fruit in f[:]:
+    preference = input("Do you like {}? ".format(fruit.lower()))
+    
+    while preference != 'yes' and preference != 'no':
+        preference = input("Please answer with yes or no. ")
+        
+    if preference == 'no':
+        f.remove(fruit)
```
Comment (charlesrobison) [inline students/thecthaeh/Lesson03/list_lab.py]: Good!

File: students/thecthaeh/Lesson03/list_lab.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+"""Series 4: Reverse the items of the list from Series 1. Remove the last item of the original list."""
+reverse_fruits = []
+
+for fruit in fruit_list[:]:
+    reverse_fruits.append(fruit[::-1])
+
+fruit_list.pop(-1)
+
+print(fruit_list)
+print(reverse_fruits)
```
Comment (charlesrobison) [inline students/thecthaeh/Lesson03/list_lab.py]: Nice job

File: students/thecthaeh/Lesson03/mailroom.py [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    prompt()
+
+#function for adding a donation
+def add_donation(donor_name, donate_amt):
+    donor_found = False
+    for donor in donor_info:
+        if donor[0] == donor_name:
+            donor[1].append(donate_amt)
+            donor_found = True
+    if donor_found == False:
```
Comment (charlesrobison) [inline students/thecthaeh/Lesson03/mailroom.py]: Should be if donor_found is False:

File: students/thecthaeh/Lesson03/mailroom.py [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    
+    #initiate the report list
+    report = []
+    
+    #fill the report with donor information
+    for donor in donor_info:
+        report.append([donor[0], sum(donor[1]), len(donor[1]), sum(donor[1])/len(donor[1])])
+    
+    #sort the report
+    sortedReport = sorted(report, key = itemgetter(1), reverse = True)
```
Comment (charlesrobison) [inline students/thecthaeh/Lesson03/mailroom.py]: No whitespaces around assignment operator: key=itemgetter(1), reverse=True

File: students/thecthaeh/Lesson03/mailroom.py [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+    prompt()
+    while actions != '3':
+        if actions == '1':
+            handle_selection1()
+            
+        elif actions == '2':
+            handle_selection2()
+            
+if __name__ == "__main__":
+    main()
```
Comment (charlesrobison) [inline students/thecthaeh/Lesson03/mailroom.py]: Works well!

File: students/thecthaeh/Lesson03/slicing.py [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+    assert reverse(a_string) == 'gnirts a si siht'
+    assert reverse(a_list) == [9, 8, 7, 6, 5, 4, 3, 2, 1]
+    assert reverse(a_tuple) == (9, 12, 4, 32, 6, 34)
+    
+    # test that new_third_order works for strings, lists, and tuples
+    assert new_third_order(a_string) == 'stringthis is a '
+    assert new_third_order(a_list) == [7, 8, 9, 1, 2, 3, 4, 5, 6]
+    assert new_third_order(a_tuple) == (12, 9, 34, 6, 32, 4)
+    
+    print("tests passed")
```
Comment (charlesrobison) [inline students/thecthaeh/Lesson03/slicing.py]: Great!

File: students/thecthaeh/Lesson03/strformat_lab.py [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+row4 = ['Ben', 6, 10500.23]
+
+table = [row1, row2, row3, row4]
+
+for row in table:
+    print("{:20}{:>20}{:>20.2f}".format(*row[:]))
+
+# Extra task
+def consecutive_tup(n):
+    print(("{:>5}" * len(n)).format(*n[:]))
```
Comment (charlesrobison) [inline students/thecthaeh/Lesson03/strformat_lab.py]: Looks good

File: students/thecthaeh/Lesson03/list_lab.py (hunk 5 above)
Comment (charlesrobison) [reply students/thecthaeh/Lesson03/list_lab.py]: Please submit this on the course site for grading

File: students/thecthaeh/Lesson03/mailroom.py (hunk 8 above)
Comment (charlesrobison) [reply students/thecthaeh/Lesson03/mailroom.py]: Please submit this on the course site for grading

File: students/thecthaeh/Lesson03/strformat_lab.py (hunk 10 above)
Comment (charlesrobison) [reply students/thecthaeh/Lesson03/strformat_lab.py]: Please submit this on the course site for grading

**Pull request merged**